File: backend/app/Models/WorkSchedule.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class WorkSchedule extends Model
{
    use HasFactory;

    protected $fillable = [
        'employee_id',
        'shift_id',
        'status_id',
        'work_date',
        'is_day_off',
    ];

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'employee_id');
    }

    public function workShift(): BelongsTo
    {
        return $this->belongsTo(WorkShift::class, 'shift_id');
    }

    public function workDateStatus(): BelongsTo
    {
        return $this->belongsTo(WorkDateStatus::class, 'status_id');
    }
}

File: hr_management_app/lib/models/work_schedule.dart
class WorkSchedule {
  final int employeeId;
  final int statusId;
  final int shiftId;
  final String workDate;
  final bool isDayOff;
  final String createdAt;
  final String updatedAt;

  WorkSchedule({
    required this.employeeId,
    required this.statusId,
    required this.shiftId,
    required this.workDate,
    required this.isDayOff,
    required this.createdAt,
    required this.updatedAt,
  });

  factory WorkSchedule.fromJson(Map<String, dynamic> json) {
    return WorkSchedule(
      employeeId: json['employee_id'] ?? 0,
      statusId: json['status_id'] ?? 0,
      shiftId: json['shift_id'] ?? 0,
      workDate: json['work_date'] ?? '',
      isDayOff: json['is_day_off'] ?? false,
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}